datalad_next/itertools/__init__.py

```diff
@@ -1,13 +1,270 @@
 """Various iterators, e.g., for subprocess pipelining and output processing
 
+This module provides iterators that are useful when combining
+iterable_subprocesses and for processing of subprocess output.
```

Comment:
Style remark: I think this is superfluous. It is largely redundant with the line above, and the additional statement about `iterable_subprocess` artificially limits the scope. From my POV these are pretty generic iterators, we plan to use them with this tools, but by themselves they do not require that.

Reply:
Removed.